src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs

```diff
@@ -61,11 +61,27 @@ internal abstract partial class AbstractNavigateToSearchService : INavigateToSea
                     (service, solutionInfo, cancellationToken) => service.SearchProjectAsync(solutionInfo, project.Id, priorityDocumentIds, searchPattern, kinds.ToImmutableArray(), cancellationToken),
                     cancellationToken).ConfigureAwait(false);
 
-                return result.HasValue ? result.Value.SelectAsArray(r => r.Rehydrate(solution)) : ImmutableArray<INavigateToSearchResult>.Empty;
+                return await RehydrateAsync(result, solution, cancellationToken).ConfigureAwait(false);
             }
 
             return await SearchProjectInCurrentProcessAsync(
                 project, priorityDocuments, searchPattern, kinds, cancellationToken).ConfigureAwait(false);
         }
+
+        private static async Task<ImmutableArray<INavigateToSearchResult>> RehydrateAsync(
+            Optional<ImmutableArray<SerializableNavigateToSearchResult>> result,
+            Solution solution,
+            CancellationToken cancellationToken)
+        {
+            if (result.HasValue)
+            {
+                var resultTasks = result.Value.SelectAsArray(r => r.RehydrateAsync(solution, cancellationToken).AsTask());
+                return (await Task.WhenAll(resultTasks).ConfigureAwait(false)).ToImmutableArray();
```

Comment:
💡 This can use the new extension as well

```suggestion
                return result.Value.SelectAsArrayAsync(r => r.RehydrateAsync(solution, cancellationToken));
```

Reply:
D'oh, good catch.